lib/libzpool/zrepl_mgmt.c

```diff
@@ -607,3 +607,49 @@ uzfs_zinfo_destroy_internal_clone(zvol_info_t *zinfo)
 	    clone_zv);
 	return (ret);
 }
+
+/*
+ * This API is used to delete stale
+ * cloned volume and backing snapshot.
+ */
+int
+uzfs_zinfo_destroy_stale_clone(zvol_info_t *zinfo)
```

Comment:
if open_dataset or hold_dataset of clone fails, its better to retry next time by releasing snap_zv and returning error.

Reply:
ok.